### ISSUE TITLE
BUG: incorrect nonce value

### ISSUE THREAD

**Issue by greged93** [lambdaclass/cairo_native#499]
I am sending a transaction to a contract which tries to get the nonce of the sender via the transaction info syscall. The below is the cairo 1 snippet I am running (with the link to the full code [here](https://github.com/kkrt-labs/kakarot-ssj/blob/main/crates/contracts/src/eoa.cairo#L89)):
```rust
assert(get_caller_address().is_zero(), 'Caller not 0');

let tx_info = get_tx_info().unbox();

let call_len = calls.len();
assert(call_len == 1, 'call len is not 1');

let call = calls.at(0);
assert(*call.to == self.kakarot_core_address(), 'to is not kakarot core');
assert!(
    *call.selector == selector!("eth_send_transaction"),
    "Validate: selector must be eth_send_transaction"
);

let signature = tx_info.signature;
let chain_id = self.chain_id();

let tx_metadata = TransactionMetadata {
     address: self.evm_address(),
     chain_id,
     account_nonce: tx_info.nonce.try_into().unwrap(),
     signature: signature
            .try_into_eth_signature(chain_id)
            .expect('signature extraction failed')
};
```

However, printing the `tx_info.nonce` value gets me a `0x1139045e000000000000000000000000000000000` instead of `0x0`. I tried tracing the nonce value all the way to the `wrap_get_execution_info_v2` and it seems the value was still correct. Couldn't figure out at what point the nonce gets set to this incorrect value.

**Comment by rodrigo-pino** [comment]
I am having a similar issue. Not located on the `nonce` but on `tx_info.account_contract_address`. This is the test I am running:

```rs
    #[external(v0)]
    fn test_get_execution_info(
        self: @ContractState,
        expected_block_info: BlockInfo,
        expected_tx_info: TxInfo,
        // Expected call info.
        expected_caller_address: felt252,
        expected_contract_address: felt252,
        expected_entry_point_selector: felt252,
    ) {
        let execution_info = starknet::get_execution_info().unbox();
        let block_info = execution_info.block_info.unbox();
        assert(block_info == expected_block_info, 'BLOCK_INFO_MISMATCH');

        let tx_info = execution_info.tx_info.unbox();
        assert(tx_info.version == expected_tx_info.version, 'VERSION_MISMATCH');
        // This one is failing
        assert(tx_info.account_contract_address == expected_tx_info.account_contract_address, 'ACC_CTRCT_ADRR_MISMATCH');
...
```

Please find the full code [here](https://github.com/NethermindEth/blockifier/blob/native/crates/blockifier/feature_contracts/cairo1/test_contract.cairo)

**Comment by azteca1998** [comment]
It seems the issue is related to the array rewrite. Basically the starknet glue code is still using the old arrays and therefore the offsets are all wrong. I'm working on it.

**Comment by azteca1998** [comment]
That PR is mostly ready: there's only a test that is failing and I'm not sure why but I'll probably fix it in a different PR.
The bug should be fixed, could you test if it fixes the bug? cc @greged93 @rodrigo-pino

**Comment by greged93** [comment]
@azteca1998 just tried but it seems like I am not getting an error with the signature. Still taking the first example, running
```rust
let signature = tx_info.signature;
(*signature.at(0)).print();
```

Seems to make the VM error out. Maybe it's important to note that my signature's length is 5. By the way I made a PR [here](https://github.com/lambdaclass/starknet_in_rust/pull/1260) to add Kakarot to SiR so that might help you with the debugging.

**Comment by azteca1998** [comment]
Well, this is embarassing. I accidentally made it so that when converting arrays to the internal representation, the start offset is always `len`, therefore it always has zero size... Fixed in #503.
